scandium-core/src/test/java/org/eclipse/californium/scandium/ConnectorHelper.java

```diff
@@ -411,17 +411,17 @@ public void clear() {
 	 * A data handler that decrements a latch on successful processing of data.
 	 *
 	 */
-	abstract class LatchDecrementingDataHandler implements DataHandler {
+	static abstract class LatchDecrementingDataHandler implements DataHandler {
 
 		private CountDownLatch latch;
 
-		public LatchDecrementingDataHandler(final CountDownLatch latch){
+		public LatchDecrementingDataHandler(CountDownLatch latch){
 			this.setLatch(latch);
 		}
 
 		@Override
-		public void handleData(InetSocketAddress endpoint, byte[] data) {
-			if (process(data)) {
+		public final void handleData(InetSocketAddress endpoint, byte[] data) {
```

Comment:
Except for very rare case, I generally don't like to much the idea to not let user override a method.

Reply:
Isn't this one of that rare cases?
The intention of the class is to count something down. The `final` ensures, that overriding is applied to the new extension point `process`, and so a decision about counting must be provided by the implementation of that new extension point. 
FMPOV, if that `handleData` is intended to be overridden in a way, that `process` doesn't allow, it should be considered to use the `DataHandler` itself instead of this `LatchDecrementingDataHandler`.